File: lib/services/question_service.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import '../models/question.dart';
import '../utils/constants.dart';

class QuestionService {
  final FirebaseFirestore _firestore = FirebaseFirestore.instance;

  // 최신 OPEN 질문 가져오기
  Future<Question?> getLatestOpenQuestion() async {
    final querySnapshot = await _firestore
        .collection(AppConstants.questionsCollection)
        .where('status', isEqualTo: AppConstants.questionStatusOpen)
        .orderBy('createdAt', descending: true)
        .limit(1)
        .get();

    if (querySnapshot.docs.isEmpty) return null;
    return Question.fromFirestore(querySnapshot.docs.first);
  }

  // 질문 실시간 스트림 (최신 1개)
  Stream<Question?> getLatestOpenQuestionStream() {
    return _firestore
        .collection(AppConstants.questionsCollection)
        .where('status', isEqualTo: AppConstants.questionStatusOpen)
        .orderBy('createdAt', descending: true)
        .limit(1)
        .snapshots()
        .map((snapshot) {
      if (snapshot.docs.isEmpty) return null;
      return Question.fromFirestore(snapshot.docs.first);
    });
  }

  // 질문 생성
  Future<String> createQuestion({
    required String text,
    required String type,
    required String optionA,
    required String optionB,
    required String deviceId,
  }) async {
    final questionRef = _firestore.collection(AppConstants.questionsCollection).doc();
    
    final question = Question(
      questionId: questionRef.id,
      text: text,
      type: type,
      optionA: optionA,
      optionB: optionB,
      votesA: 0,
      votesB: 0,
      totalVotes: 0,
      createdAt: DateTime.now(),
      status: AppConstants.questionStatusOpen,
      createdBy: deviceId,
    );

    await questionRef.set(question.toFirestore());
    return questionRef.id;
  }

  // 질문 가져오기
  Future<Question?> getQuestion(String questionId) async {
    final doc = await _firestore
        .collection(AppConstants.questionsCollection)
        .doc(questionId)
        .get();

    if (!doc.exists) return null;
    return Question.fromFirestore(doc);
  }

  // 질문 실시간 스트림
  Stream<Question?> getQuestionStream(String questionId) {
    return _firestore
        .collection(AppConstants.questionsCollection)
        .doc(questionId)
        .snapshots()
        .map((doc) => doc.exists ? Question.fromFirestore(doc) : null);
  }
}

File: src/services/questionService.js
// 로컬 스토리지 기반으로 단순화
import * as localStorage from './localStorage';

export const getLatestOpenQuestion = localStorage.getLatestOpenQuestion;
export const createQuestion = localStorage.createQuestion;
export const getQuestion = localStorage.getQuestion;

export const subscribeToLatestOpenQuestion = (callback) => {
  // 간단한 폴링 방식
  const interval = setInterval(async () => {
    const question = await localStorage.getLatestOpenQuestion();
    callback(question);
  }, 1000);
  
  // 초기값도 전달
  localStorage.getLatestOpenQuestion().then(callback);
  
  return () => clearInterval(interval);
};

export const subscribeToQuestion = (questionId, callback) => {
  // 간단한 폴링 방식
  const interval = setInterval(async () => {
    const question = await localStorage.getQuestion(questionId);
    callback(question);
  }, 1000);
  
  // 초기값도 전달
  localStorage.getQuestion(questionId).then(callback);
  
  return () => clearInterval(interval);
};
